src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts

```diff
@@ -439,12 +441,11 @@ export class SavePlanFile extends Action {
 	public override async run(context: ExecutionPlanView): Promise<void> {
 		const workspaceFolders = await context.workspaceContextService.getWorkspace().folders;
 		const defaultFileName = 'plan';
-		let currentWorkSpaceFolder: URI;
+		let defaultSaveUri: URI;
 		if (workspaceFolders.length !== 0) {
-			currentWorkSpaceFolder = workspaceFolders[0].uri;
-			currentWorkSpaceFolder = URI.joinPath(currentWorkSpaceFolder, defaultFileName); //appending default file name to workspace uri
+			defaultSaveUri = URI.joinPath(workspaceFolders[0].uri, defaultFileName); //appending default file name to workspace uri
 		} else {
-			currentWorkSpaceFolder = URI.parse(defaultFileName); // giving default name
+			defaultSaveUri = URI.file(path.join(os.homedir(), defaultFileName)); // giving default name
```

Comment:
I would suggest using [fileDialogService.defaultFilePath(Schemas.file)](https://github.com/Microsoft/azuredatastudio/blob/9a14fef8075965f62c2d4efdfa1a30bf6ddddcf9/src/vs/platform/dialogs/common/dialogs.ts#L312)

This way it'll pick the last save location if the user previously saved a file elsewhere, which is usually desirable behavior.

This will update anytime a user saves ANY file though, so a further improvement could be to also store the save path every time the user saves a sqlplan and then check that first. See https://github.com/Microsoft/azuredatastudio/blob/9a14fef8075965f62c2d4efdfa1a30bf6ddddcf9/src/vs/workbench/contrib/files/browser/fileImportExport.ts#L799 for an example.

Reply:
Good idea - have added the path persistence logic 👍